backend/src/main/java/org/neuefische/applicationmangementapp/service/ApplicationService.java

```diff
@@ -1,26 +1,44 @@
 package org.neuefische.applicationmangementapp.service;
 
+import org.neuefische.applicationmangementapp.execaptions.NoSuchId;
 import org.neuefische.applicationmangementapp.model.Application;
+
+import org.neuefische.applicationmangementapp.model.ApplicationDtoForCreated;
+import org.neuefische.applicationmangementapp.model.ApplicationDtoForEdit;
+import org.neuefische.applicationmangementapp.model.appliStatus;
 import org.neuefische.applicationmangementapp.repo.ApplicationRepo;
-import org.springframework.beans.factory.annotation.Autowired;
+
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
 import java.util.List;
+import java.util.Optional;
+
 @Service
 public class ApplicationService {
 
-    final private ApplicationRepo applicationRepo;
+    private final  ApplicationRepo applicationRepo;
 
     public ApplicationService(ApplicationRepo applicationRepo) {
         this.applicationRepo = applicationRepo;
     }
 
-    public Application addApplication(Application application) {
+    public Application addApplication(ApplicationDtoForCreated applicationDto) {
+        Application application=new Application(IdService.getId(),applicationDto.jobOfferID(),applicationDto.resume(),applicationDto.coverLetter(), appliStatus.OPEN,applicationDto.reminderTime(), LocalDate.now());
      return    applicationRepo.save(application);
     }
 
-    public Application updateApplication(Application application) {
-     return    applicationRepo.save(application);
+    public Application updateApplication(String id, ApplicationDtoForEdit applicationDto) throws NoSuchId {
+
+       Optional<Application> oApplication= applicationRepo.findById(id);
+
+        if (oApplication.isEmpty()){
+            throw new NoSuchId("no such id: "+id );
+        }else{
```

Comment:
```suggestion
        } else {
```

I think an autoformatter would be really useful here 😉